tests/integration/telemetry/stats/prometheus/TCP/stats_tcp_filter_test.go

```diff
@@ -0,0 +1,194 @@
+// Copyright 2019 Istio Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package TCP
+
+import (
+	"testing"
+
+	"fmt"
+	"time"
+
+	"istio.io/istio/pkg/config/mesh"
+	"istio.io/istio/pkg/test/framework"
+	"istio.io/istio/pkg/test/framework/components/bookinfo"
+	"istio.io/istio/pkg/test/framework/components/environment"
+	"istio.io/istio/pkg/test/framework/components/galley"
+	"istio.io/istio/pkg/test/framework/components/ingress"
+	"istio.io/istio/pkg/test/framework/components/istio"
+	"istio.io/istio/pkg/test/framework/components/mixer"
+	"istio.io/istio/pkg/test/framework/components/namespace"
+	"istio.io/istio/pkg/test/framework/components/prometheus"
+	"istio.io/istio/pkg/test/framework/label"
+	"istio.io/istio/pkg/test/framework/resource"
+	"istio.io/istio/pkg/test/util/file"
+	"istio.io/istio/pkg/test/util/retry"
+	util "istio.io/istio/tests/integration/mixer"
+	util_prometheus "istio.io/istio/tests/integration/telemetry/stats/prometheus"
+)
+
+const (
+	metadataExchangeFilterConfig = "testdata/metadata_exchange_tcp_filter.yaml"
+	statsFilterConfig            = "testdata/stats_filter.yaml"
+	cleanupFilterConfig          = "testdata/cleanup.yaml"
+)
+
+var (
+	ist        istio.Instance
+	bookinfoNs namespace.Instance
+	g          galley.Instance
+	ing        ingress.Instance
+	prom       prometheus.Instance
+)
+
+func TestTcpMetric(t *testing.T) { // nolint:interfacer
+	framework.
+		NewTest(t).
+		RequiresEnvironment(environment.Kube).
+		Run(func(ctx framework.TestContext) {
+			addr := ing.HTTPAddress()
+			url := fmt.Sprintf("http://%s/productpage", addr.String())
+			g.ApplyConfigOrFail(
+				t,
+				bookinfoNs,
+				bookinfo.GetDestinationRuleConfigFileOrFail(t, ctx).LoadWithNamespaceOrFail(t, bookinfoNs.Name()),
+				bookinfo.NetworkingTCPDbRule.LoadWithNamespaceOrFail(t, bookinfoNs.Name()),
+			)
+			defer g.DeleteConfig(
+				bookinfoNs,
+				bookinfo.GetDestinationRuleConfigFileOrFail(t, ctx).LoadWithNamespaceOrFail(t, bookinfoNs.Name()),
+				bookinfo.NetworkingTCPDbRule.LoadWithNamespaceOrFail(t, bookinfoNs.Name()),
+			)
+
+			systemNM := namespace.ClaimSystemNamespaceOrFail(ctx, ctx)
+			mxc, err := file.AsString(metadataExchangeFilterConfig)
```

Comment:
Please add a todo here as well to install these filters via istioctl.

Reply:
Done